### ISSUE TITLE
[java] SingleMethodSingleton false positives with inner classes

### ISSUE THREAD

**Issue by TWiStErRob** [pmd/pmd#816]
**Affects PMD Version:** 6.0.0

**Rule:** [`SingleMethodSingleton`](https://pmd.github.io/pmd-6.0.0/pmd_rules_java_errorprone.html#singlemethodsingleton)

**Description:** `SingleMethodSingleton` [uses a single `Set`](https://github.com/pmd/pmd/blob/pmd_releases/6.0.0/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/SingleMethodSingletonRule.java#L31) to determine if any duplication happened. But while traversing the AST multiple classes may be visited. I think that needs to be a `Map<Class, Method>`. (Also it should only warn for `static` methods.)

**Code Sample demonstrating the issue:**

```java
// real example that triggered this report
final class Outer {
  private static final Outer INSTANCE = new Outer();
  private Outer() {}
  static Outer getInstance() { return INSTANCE; }

  static class Factory {
    private static final Factory FACTORY = new Factory();
    private Factory() { }
    static Factory getInstance() { return FACTORY; }
  }
}

// another possible scenario
final class Siblings {
  static class Inner1 {
    static Inner1 getInstance() { return null; }
  }
  static class Inner2 {
    static Inner2 getInstance() { return null; }
  }
}
```
> Outer.java | 4 | Class contains multiple getInstance methods. Please review.
> Outer.java | 9 | Class contains multiple getInstance methods. Please review.
> Siblings.java | 3 | Class contains multiple getInstance methods. Please review.
> Siblings.java | 6 | Class contains multiple getInstance methods. Please review.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*: Gradle

**Comment by jsotuyod** [comment]
@TWiStErRob nice find! Definitely an issue, and an easy one shall anyone want to submit a first PR.

We will probably tackle this on 6.1.0 otherwise.
